packages/pigeon/tool/shared/flutter_utils.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:convert';
+import 'dart:io';
+
+String getFlutterCommand() => Platform.isWindows ? 'flutter.bat' : 'flutter';
+
+/// Returns the first device listed by `flutter devices` that targets
+/// [platform], or null if there is no such device.
+Future<String?> getDeviceForPlatform(String platform) async {
```

Comment:
This code is duplicated from the repo tooling (with minor changes). While it's not ideal to duplicate it, it's not much code, and restructuring everything to allow using the repo tooling directly on Pigeon's tests (and thus eliminating most of run_tests.*) is a much bigger project that I don't want to gate other work.